### PR TITLE
[state sync] spawn_blocking around synchronous executor calls

### PR DESCRIPTION
**Pull request by msmouse**
### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3494)

### CODE REVIEW

File: state-sync/state-sync-v2/state-sync-driver/src/storage_synchronizer.rs [hunk 1]
```diff
@@ -428,11 +428,18 @@ fn spawn_executor<ChunkExecutor: ChunkExecutorTrait + 'static>(
...
                     );
                     let num_outputs = outputs_with_proof.transactions_and_outputs.len();
-                    let result = chunk_executor.apply_chunk(
-                        outputs_with_proof,
-                        &target_ledger_info,
-                        end_of_epoch_ledger_info.as_ref(),
-                    );
+                    let chunk_executor_clone = chunk_executor.clone();
+                    // `spawn_blocking` so that the heavy synchronous function call doesn't
+                    // block the aysnc thread.
```
Comment (JoshLind) [inline state-sync/state-sync-v2/state-sync-driver/src/storage_synchronizer.rs:433]: nit: "aysnc" -> "async"

File: state-sync/state-sync-v2/state-sync-driver/src/storage_synchronizer.rs [hunk 2]
```diff
@@ -515,7 +522,14 @@ fn spawn_committer<
                 &metrics::STORAGE_SYNCHRONIZER_LATENCIES,
                 metrics::STORAGE_SYNCHRONIZER_COMMIT_CHUNK,
             );
-            match chunk_executor.commit_chunk() {
+            let chunk_executor_clone = chunk_executor.clone();
+            let commit_fut = move || chunk_executor_clone.commit_chunk();
+            // `spawn_blocking` so that the heavy synchronous function call doesn't
+            // block the aysnc thread.
```
Comment (JoshLind) [inline state-sync/state-sync-v2/state-sync-driver/src/storage_synchronizer.rs:528]: nit: "aysnc" -> "async"

**Review by JoshLind — APPROVED** [review]
Thanks @msmouse! LGTM 😄 Hopefully forge reports good numbers so we're good to go.

File: state-sync/state-sync-v2/state-sync-driver/src/storage_synchronizer.rs (hunk 1 above)
Comment (msmouse) [reply state-sync/state-sync-v2/state-sync-driver/src/storage_synchronizer.rs:433]: ... sorry, moving (me) fast is hard

**Review by grao1991 — APPROVED** [review]

**Review by igor-aptos — APPROVED** [review]

**Pull request merged**